src/vm/tests/simple_apply_eval.rs

```diff
@@ -541,6 +541,173 @@ fn test_simple_arithmetic_functions() {
+        CheckErrors::UnionTypeValueError(
+            vec![TypeSignature::IntType, TypeSignature::UIntType],
+            Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                data: "baa".as_bytes().to_vec(),
+            }))),
+        )
+        .into(),
+        CheckErrors::UnionTypeValueError(
+            vec![TypeSignature::IntType, TypeSignature::UIntType],
+            Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                data: "baa".as_bytes().to_vec(),
+            }))),
+        )
+        .into(),
+        CheckErrors::UnionTypeValueError(
+            vec![TypeSignature::IntType, TypeSignature::UIntType],
+            Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                data: "baa".as_bytes().to_vec(),
+            }))),
+        )
+        .into(),
+        CheckErrors::UnionTypeValueError(
+            vec![TypeSignature::IntType, TypeSignature::UIntType],
+            Value::Sequence(SequenceData::String(CharType::ASCII(ASCIIData {
+                data: "baa".as_bytes().to_vec(),
+            }))),
+        )
+        .into(),
+    ];
+
+    // Note: Execute against Clarity1.
+    error_tests
+        .iter()
+        .zip(error_expectations)
+        .for_each(|(program, expectation)| {
+            assert_eq!(*expectation, vm_execute(program).unwrap_err())
+        });
+}
+
+#[test]
+fn test_sequence_comparisons_v2() {
+    // Tests the sequence comparisons against ClarityVersion2. The new kinds of
+    // sequence comparison *should* work.
+    let success_tests = [
+        ("(is-eq \"aaa\" \"aaa\")", Value::Bool(true)),
+        ("(is-eq \"aba\" \"aaa\")", Value::Bool(false)),
+        ("(is-eq u\"aaa\" u\"aaa\")", Value::Bool(true)),
+        ("(is-eq 0x010203 0x010203)", Value::Bool(true)),
+        ("(is-eq 0x090203 0x010203)", Value::Bool(false)),
+        ("(> \"baa\" \"aaa\")", Value::Bool(true)),
+        ("(< \"baa\" \"aaa\")", Value::Bool(false)),
+        ("(>= \"baa\" \"aaa\")", Value::Bool(true)),
+        ("(>= \"aaa\" \"aaa\")", Value::Bool(true)),
+        ("(< \"aaa\" \"baa\")", Value::Bool(true)),
+        ("(> \"aaa\" \"baa\")", Value::Bool(false)),
+        ("(<= \"aaa\" \"baa\")", Value::Bool(true)),
+        ("(<= \"aaa\" \"aaa\")", Value::Bool(true)),
+        ("(> u\"baa\" u\"aaa\")", Value::Bool(true)),
+        ("(>= u\"baa\" u\"aaa\")", Value::Bool(true)),
+        ("(>= u\"aaa\" u\"aaa\")", Value::Bool(true)),
+        ("(< u\"aaa\" u\"baa\")", Value::Bool(true)),
+        ("(<= u\"aaa\" u\"baa\")", Value::Bool(true)),
+        ("(<= u\"aaa\" u\"aaa\")", Value::Bool(true)),
+        ("(> 0x0200 0x0100)", Value::Bool(true)),
+        ("(>= 0x0200 0x0100)", Value::Bool(true)),
+        ("(>= 0x0100 0x0100)", Value::Bool(true)),
+        ("(< 0x0100 0x0200)", Value::Bool(true)),
+        ("(<= 0x0100 0x0200)", Value::Bool(true)),
+        ("(<= 0x0100 0x0100)", Value::Bool(true)),
```

Comment:
Can you expand test coverage to include strings and buffers of _different_ lengths?  The `PartialOrd` trait is automatically derived for `Vec<T>` if it is implemented for `T`, so since `T` is either `u8` or `Vec<u8>` in this case, we should expect the following to be true:

`(< "" "a")`
`(> "a" "")`
`(<= "" "a")`
`(>= u"a" u"")`
`(< u"" u"a")`
`(> u"a" u"")`
`(<= u"" u"a")`
`(>= u"a" u"")`
`(< 0x 0x01)`
`(> 0x01 0x)`
`(<= 0x 0x01)`
`(>= 0x01 0x)`

Reply:
I updated the tests, to 1) use different lengths, and 2) make the set of tests comprehensive.